dffml/high_level/ml.py

```diff
@@ -125,6 +137,13 @@ async def accuracy(
     Accuracy: 0.0
     """
     async with contextlib.AsyncExitStack() as astack:
+        # if predict feature(s) were passed into the model, extract and set args for sources
+        if isinstance(args[0], (Feature, Features)):
+            features = args[0]
+            args = args[1:]
+        elif isinstance(args[-1], (Feature, Features)):
+            features = args[0]
+            args = args[:-1]
```

Comment:
We have overloaded the possible types of the variable length arguments too far with this implementation. The existing logic is not very clean.

Let us instead modify the function signature of `accuracy()` to accept either a `Feature`, or `Features` as the third argument.

```python
async def accuracy(
    model,
    accuracy_scorer: Union[AccuracyScorer, AccuracyContext],
    features: Union[Feature, Features],
    *args: Union[BaseSource, Record, Dict[str, Any], Feature, Features],
) -> float:
```

```suggestion
    # TODO Use this to ensure that we're always passing features before records
    # We can remove it eventually once we know we've updated everywhere
    # appropriately
    if not isinstance(features, Feature):
        raise TypeError(f"features was {type(features)}: {features!r}. Should have been Feature or Features")
    if isinstance(features, Feature):
        features = Features(features)
    async with contextlib.AsyncExitStack() as astack:
```